### PR TITLE
Add project definition version to telemetry

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Added `definition_version` information in telemetry dict. This required to change `cli_context.project_definition` to be a full project definition instead of streamlit/snowpark/native_app section.

### CODE REVIEW

File: src/snowflake/cli/app/telemetry.py [hunk 1]
```diff
@@ -74,9 +76,18 @@ def _find_command_info() -> TelemetryDict:
...
+        CLITelemetryField.PROJECT_DEFINITION_VERSION: _get_definition_version(),
     }
 
 
+def _get_definition_version() -> str:
+    from snowflake.cli.api.cli_global_context import cli_context
+
+    if cli_context.project_definition:
+        return cli_context.project_definition.definition_version
+    return "no_definition"
```
Comment (sfc-gh-cgorrie) [inline src/snowflake/cli/app/telemetry.py:88]: Should it be `None` instead?

File: src/snowflake/cli/app/telemetry.py [hunk 2]
```diff
@@ -74,9 +76,18 @@ def _find_command_info() -> TelemetryDict:
         CLITelemetryField.COMMAND_OUTPUT_TYPE: ctx.params.get(
             "format", OutputFormat.TABLE
         ).value,
+        CLITelemetryField.PROJECT_DEFINITION_VERSION: _get_definition_version(),
     }
 
 
+def _get_definition_version() -> str:
```
Comment (sfc-gh-cgorrie) [inline src/snowflake/cli/app/telemetry.py:83]:
```suggestion
def _get_definition_version() -> Optional[str]:
```

File: src/snowflake/cli/app/telemetry.py (hunk 1 above)
Comment (sfc-gh-turbaszek) [reply src/snowflake/cli/app/telemetry.py:88]: I decided to use a more meaningful string, then when you build a dashboard you can see table `1 | 1.1 | 2.0 | no_definition`. Thought this is nicer. WDYT?

**Review by sfc-gh-cgorrie — APPROVED** [review]
LGTM, though I haven't run through manual testing.

**Comment by sfc-gh-cgorrie** [comment]
Just ran through manual testing (`snow app run`, test, `snow app teardown`). All good

**Pull request merged**